network/src/peer/mod.rs

```diff
@@ -343,17 +335,32 @@ where
                         );
                         err
                     })?;
-                Ok(())
+            }
+            NetworkMessage::Error(error) => {
+                warn!(
+                    "Peer {} sent an error message: {:?}",
+                    self.peer_id().short_str(),
+                    error,
+                );
             }
             NetworkMessage::Ping(nonce) => {
                 let pong = NetworkMessage::Pong(nonce);
                 let (ack_tx, _) = oneshot::channel();
                 // Resond to a ping right away.
                 write_reqs_tx.send((pong, ack_tx)).await?;
-                Ok(())
             }
-            _ => unreachable!("Unhandled"),
-        }
+            NetworkMessage::Pong(_nonce) => {
+                // TODO(davidiw) forward to a pong pub/sub and move HealthChecker to leverage this
```

Comment:
Let's start putting TODOs with issues associated with them!

Reply:
I plan on taking care of this one rather quickly :P but I agree